woocommerce/woocommerce-ios
# Shipping Label M2: add new package screen that works like a middleware

Closes #3911 

## Description
As part of M2, I implemented a screen that works like a middleware, and that for the moment will be useful in the next PR for showing the list of Service Packages, but in the future (in the M4) will be useful also for showing the screen for adding a Custom Package. So, I implemented also a reusable component called `SegmentedView` which will be shown on top of the view (for the moment is hidden) for selecting the custom package or the service package menu.

## Testing
1. Launch the app in debug mode (because of the feature flag).
2. Open an order detail.
3. Press the button "Create Shipping Label".
4. Press the button "Continue" in Create Shipping Label form, under the **Ship From** cell. Confirm the address.
5. Press the button "Continue" in Create Shipping Label form, under the **Ship To** cell. Confirm the address.
6. Press the button "Continue" in Create Shipping Label form, under the **Package Details** cell.
7. You should see the new screen that shows the Package Details screen and all the items that will be shipped.
8. Press the "Package selected" row. -> You should see the new empty screen "Add New Package".
9. Remove the `.hidden()` method in `ShippingLabelAddNewPackage.swift`, re-launch the app, and redo the entire testing procedure. -> This time in the Add New Package screen you should see the `SegmentedView` on top.

## Screenshots

| Add New Package container view             |  Add New Package container view with SegmentedView displayed |
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 12 Pro - 2021-04-12 at 17 09 58](https://user-images.githubusercontent.com/495617/114417519-f3fea280-9bb1-11eb-8d26-8bb073c5491c.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-04-12 at 17 13 36](https://user-images.githubusercontent.com/495617/114418458-ce25cd80-9bb2-11eb-82f5-1901ba80d9fb.png)




Update release notes:
